pkg/reconciler/pipeline/controller.go

```diff
@@ -140,9 +140,11 @@ func (r *pipelineReconciler) reconcileNonLifecycleChanges(ctx context.Context, p
 	pl.Status.InitConditions()
 	if err := ValidatePipeline(pl); err != nil {
 		log.Errorw("Validation failed", zap.Error(err))
+		pl.Status.ResetTopologyCounts()
```

Comment:
This means if it's a failure caused by an update, we also reset all the numbers... There's nothing wrong with it, i'm just thinking, if keeping the numbers is better?

Reply:
yeah that's what it's doing, i wasn't sure if we wanted to clear them or not.  I'll go ahead and remove the reset.